### PR TITLE
Added a verbose validation for Tip component

### PR DESCRIPTION
**Pull request by stanislav-slavov**
ref #619

### CODE REVIEW

File: src/popup/router/pages/Tip.vue [hunk 1]
```diff
@@ -145,11 +112,52 @@ export default {
...
+        }
+        if (this.selectedToken && !this.tippingV2) {
+          return { error: true, msg: this.$t('pages.tipPage.v1FungibleTokenTipError') };
+        }
+        if (!this.selectedToken && (!this.amount || this.amount < minTipAmount)) {
+          return { error: true, msg: this.$t('pages.tipPage.minAmountError') };
+        }
+        if (this.selectedToken && !this.amount) {
+          return { error: true, msg: this.$t('pages.tipPage.requiredAmountError') };
+        }
```
Comment (davidyuk) [inline src/popup/router/pages/Tip.vue:137]:
```suggestion
        if (!this.amount) {
          return { error: true, msg: this.$t('pages.tipPage.requiredAmountError') };
        }
        if (!this.selectedToken && this.amount < minTipAmount) {
          return { error: true, msg: this.$t('pages.tipPage.minAmountError') };
        }
```

File: src/popup/router/components/TokenAmount.vue [hunk 2]
```diff
@@ -12,12 +12,12 @@ import { mapState } from 'vuex';
 
 export default {
   props: {
-    amount: { type: Number, required: true },
+    amount: { type: [Number, String], required: true },
```
Comment (davidyuk) [inline src/popup/router/components/TokenAmount.vue:15]: I would prefer a more strict validation. Add the explicit conversion to number where it is needed instead.

File: src/popup/router/pages/Tip.vue [hunk 3]
```diff
@@ -24,47 +24,17 @@
...
-            !url ||
-            urlStatus === 'blacklisted' ||
-            note.length > 280
-          "
-          bold
-          data-cy="send-tip"
-        >
+        <AmountSend v-model="amount" />
+        <Textarea v-model="note" :placeholder="$t('pages.tipPage.titlePlaceholder')" size="sm" />
+        <div class="validation-msg" :class="{ show: validationStatus.msg }">
```
Comment (davidyuk) [inline src/popup/router/pages/Tip.vue:34]:
```suggestion
        <div class="validation-msg">
```
With empty content it is anyway is not visible.

File: src/popup/router/components/TokenAmount.vue [hunk 4]
```diff
@@ -17,7 +17,7 @@ export default {
   },
   computed: {
     amountRounded() {
-      return +this.amount.toFixed(2);
+      return this.amount.toFixed(2);
```
Comment (davidyuk) [inline src/popup/router/components/TokenAmount.vue:20]: This is needed to remove trailing zeroes, e.g. "1.10" => "1.1"

**Review by davidyuk — APPROVED** [review]

**Review by mradkov — APPROVED** [review]

**Pull request merged**